### DISCUSSION TITLE
FundMe lesson : TypeError in withdraw function()

### DISCUSSION THREAD

**Discussion by AyuK03** [smartcontractkit/full-blockchain-solidity-course-py#1741] [Q&A] [answered]
` // SPDX-License-Identifier: MIT

pragma solidity >=0.8.9 <0.9.0;

import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
import "@chainlink/contracts/src/v0.6/vendor/SafeMathChainlink.sol";
... more code here same as told in the Fund me lesson ...
   function withdraw() payable public{
        msg.sender.transfer(address(this).balance);
    }`

I am getting an error with the withdraw function as follows:

![db fundme](https://user-images.githubusercontent.com/99130870/195116127-7f494f07-5841-4d48-8a29-ffd82db8ea1c.png)

from solidity:
TypeError: "send" and "transfer" are only available for objects of type "address payable", not "address".
  --> contracts/Ak_FundMe.sol:63:9:
   |
63 |         msg.sender.transfer(address(this).balance);
   |         ^^^^^^^^^^^^^^^^^^^

Please guide.

**Comment by cromewar** [comment, marked as answer]
hello @AyuK03 
This is because you are trying to use solidity version 0.8, and there are some changes on that version, you have two solutions here.
1. Back to 0.6 to match the course.
2. make.sender payable `payable(msg.sender).transfer....`.

I hope this info might help.
